src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -84,10 +88,13 @@ internal async override Task PreRunAsync()
 
         internal override async Task<AuthenticationResult> ExecuteAsync(CancellationToken cancellationToken)
         {
-           
             var logger = AuthenticationRequestParameters.RequestContext.Logger;
             MsalAccessTokenCacheItem cachedAccessTokenItem = null;
-
+            if (AuthenticationRequestParameters.IsBrokerEnabled)
```

Comment:
What about iOS? Will it still work?

Reply:
Please see comment above. It should be obvious from reading the code the difference between android and ios brokers.